src/modules/drop-erc-721.ts

```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
+   * @returns The NFT metadata for all NFTs queried.
+   */
+  public async getAllClaimed(
+    queryParams?: QueryAllParams,
+  ): Promise<NFTMetadataOwner[]> {
+    const start = BigNumber.from(queryParams?.start || 0).toNumber();
+    const count = BigNumber.from(
+      queryParams?.count || DEFAULT_QUERY_ALL_COUNT,
+    ).toNumber();
+    const maxId = Math.min(
+      (await this.contractWrapper.readContract.nextTokenIdToClaim()).toNumber(),
+      start + count,
+    );
+    return await Promise.all(
+      Array.from(Array(maxId).keys()).map((i) => this.get(i.toString())),
+    );
+  }
+
+  /**
+   * Get All Unclaimed NFTs
+   *
+   * @param queryParams - optional filtering to only fetch a subset of results.
+   * @returns The NFT metadata for all NFTs queried.
+   */
+  public async getAllUnclaimed(
+    queryParams?: QueryAllParams,
+  ): Promise<NFTMetadata[]> {
+    const start = BigNumber.from(queryParams?.start || 0).toNumber();
+    const count = BigNumber.from(
+      queryParams?.count || DEFAULT_QUERY_ALL_COUNT,
+    ).toNumber();
+    const maxId = BigNumber.from(
+      Math.min(
+        (
+          await this.contractWrapper.readContract.nextTokenIdToMint()
+        ).toNumber(),
+        start + count,
+      ),
+    );
+    const unmintedId =
+      await this.contractWrapper.readContract.nextTokenIdToClaim();
+    return await Promise.all(
+      Array.from(Array(maxId.sub(unmintedId).toNumber()).keys()).map((i) =>
+        this.getTokenMetadata(unmintedId.add(i).toString()),
+      ),
+    );
+  }
+
+  /**
+   * Get Owned NFTs
+   *
+   * @remarks Get all the data associated with the NFTs owned by a specific wallet.
+   *
+   * @example
+   * ```javascript
+   * // Address of the wallet to get the NFTs of
+   * const address = "{{wallet_address}}";
+   * const nfts = await module.getOwned(address);
+   * console.log(nfts);
+   * ```
+   *
+   * @returns The NFT metadata for all NFTs in the module.
+   */
+  public async getOwned(_address?: string): Promise<NFTMetadataOwner[]> {
+    const address = _address
+      ? _address
+      : await this.contractWrapper.getSignerAddress();
+    const balance = await this.contractWrapper.readContract.balanceOf(address);
+    const indices = Array.from(Array(balance.toNumber()).keys());
+    const tokenIds = await Promise.all(
+      indices.map((i) =>
+        this.contractWrapper.readContract.tokenOfOwnerByIndex(address, i),
+      ),
+    );
+    return await Promise.all(
+      tokenIds.map((tokenId) => this.get(tokenId.toString())),
+    );
+  }
+
+  /**
+   * Get the current owner of a given NFT within this Drop
+   *
+   * @param tokenId - the tokenId of the NFT
+   * @returns the address of the owner
+   */
+  public async ownerOf(tokenId: string): Promise<string> {
+    return await this.contractWrapper.readContract.ownerOf(tokenId);
+  }
+
+  // TODO extract all the claim condition logic to its own class
```

Comment:
Are claim conditions reused in other modules? I'm gonna extract it out I think, it has a lot of logic, would be another namespace like `module.claimConditons.getActive()`

lmk what you think @atbe

Reply:
claim conditions exist on erc721 drop and erc1155drop (for now), they work slightly different in both cases but we might be able to extract them out